crates/sequencer/src/sequencer.rs

```diff
@@ -769,6 +824,39 @@ where
         });
     }
 
+    pub async fn restore_mempool(
+        &self,
+        db_config: SharedBackupDbConfig,
+    ) -> Result<(), anyhow::Error> {
+        match PostgresConnector::new(db_config).await {
+            Ok(pg_connector) => {
+                let mempool_txs = pg_connector.get_all_txs().await?;
+                for tx in mempool_txs {
+                    // TODO Handle error
+                    let decoded = TransactionSigned::decode(&mut tx.tx.as_slice()).unwrap();
+                    // TODO Handle error
+                    let signed_ec_recovered = decoded.into_ecrecovered().unwrap();
+
+                    let length_without_header = signed_ec_recovered.length_without_header();
+                    let pooled_tx =
+                        EthPooledTransaction::new(signed_ec_recovered, length_without_header);
+
+                    // TODO Handle error
+                    let _ = self
+                        .mempool
+                        .add_external_transaction(pooled_tx)
+                        .await
+                        .unwrap();
+                }
```

Comment:
why so different from in rpc?

Reply:
turns I was using the wrong encoding when I should have used encode_enveloped and same for decoding.
Fixed it